### PR TITLE
chore: enhance digest system with comprehensive testing

### PR DESCRIPTION
**Pull request by MicBun**
Implement comprehensive edge case testing and fix digest system issues to ensure production reliability:

**Major Fixes:**
- Resolve idempotency violation by adding already-processed day detection in batch_digest
- Fix empty array handling with COALESCE to prevent NULL returns

**Test Enhancements:**
- Add comprehensive flag verification in testOtherCombinationFlagPermutations with proper mathematical expectations

**Technical Improvements:**
- Implement verifyCorrectRecordsPersisted() function for duplicate resolution validation
- Implement verifyMultipleDuplicatesCorrectRecords() function for multiple duplicate scenarios
- Implement verifySpecificFlags() function for flag combination testing
- Add proper cleanup logic for already-processed days in idempotency scenarios

**Edge Cases Covered:**
- Boundary timestamp semantics with [day_start, day_end) interval logic
- Multiple duplicate record resolution with created_at tie-breaking (2-5 duplicates per timestamp)
- Flag combination mathematics (OPEN+HIGH=3, OPEN+LOW=5, impossible combinations explained)
- Empty array parameter handling across all digest functions
- Idempotency guarantees for reprocessing same day data

These changes ensure the digest system handles all production edge cases correctly and provides comprehensive test coverage for data integrity validation.

resolves: https://github.com/trufnetwork/node/issues/1129

## Summary by CodeRabbit

- New Features
  - Idempotent, bulk digest processing with marker upserts and preserved OHLC markers; returns totals for preserved/deleted items and processed days.

- Performance
  - Bulk aggregation and capped deletions with smarter batching for higher throughput and scalable pruning.

- Bug Fixes
  - NULL-safe length handling and more robust duplicate/boundary/timestamp semantics; safer gating when no inputs exist.

- Tests
  - Expanded edge-case test suite for duplicates, boundary timestamps, partial deletes, single-record days, and idempotency.

### CODE REVIEW

File: internal/migrations/020-digest-actions.sql [hunk 1]
```diff
@@ -87,18 +87,82 @@ CREATE OR REPLACE ACTION batch_digest(
...
 
-    -- Early exit if no candidates provided
-    if array_length($valid_stream_refs) = 0 {
-        RETURN 0, 0, 0, false;
+    for $candidate in
+    WITH candidates AS (
+        SELECT sr, di
+        FROM UNNEST($stream_refs, $day_indexes) AS u(sr, di)
+    ),
+    candidate_status AS (
```
Comment (outerlook) [inline internal/migrations/020-digest-actions.sql:100]: I'm a bit concerned about idempotency here, imagine this scenario:

1) insert two records on the same day at different times: earlier t1=200, later t2=100.
2) run `batch_digest` → OHLC is [200, 200, 100, 100].
3) insert a new record at the same event_time as the later one (t2) with value 250 and a higher created_at.
4) run `batch_digest` again.
5) expect OHLC to become [200, 250, 200, 250] — the duplicate at t2 should supersede the old value because of the higher created_at.

My worry is that if `batch_digest` idempotency only checks whether the day is "already processed" (and doesn’t account for created_at-based duplicates on the same event_time), step 4 becomes a no-op and we miss the update.

I do think the idempotency test is important, but I'm also thinking on where we draw the line on who enforces it.

What do you think if we:
- Keep `batch_digest` straightforward: when we ask it to reprocess specific days, it should fully recompute them and apply any changes (avoid input short-circuiting beyond correctness).
- Handle idempotency at the scheduler level (`auto_digest`). If there’s nothing to do, `auto_digest` simply won’t schedule that day again (pending/prune set is empty), so we avoid redundant work anyway.

but also has the new tests that make sure that running twice still leads to a correct result (and if we run the `auto_digest` it will not reprocess the day again)

Comment (MicBun) [reply internal/migrations/020-digest-actions.sql:100]: Let me try the non short-circuiting ones

Comment (MicBun) [reply internal/migrations/020-digest-actions.sql]: I made it non short-circuting @outerlook

File: internal/migrations/020-digest-actions.sql [hunk 2]
```diff
@@ -296,7 +289,7 @@ CREATE OR REPLACE ACTION batch_digest(
         }
 
         -- Step 2b: BULK DELETION using aligned arrays (guaranteed alignment)
-        if array_length($agg_stream_refs) > 0 {
+        if COALESCE(array_length($agg_stream_refs), 0) > 0 {
```
[the remark this answers is not in the document]
Comment (MicBun) [reply internal/migrations/020-digest-actions.sql:293]: not needed
